### PR TITLE
Disable touch on nodes w/o children

### PR DESCRIPTION
**Pull request by brightsider**

### CODE REVIEW

**Comment by zaguiini** [comment]
Thanks for your contribution!

I'm not sure if I agree with it, though. Why are we disabling the touch of the node? The user might have a callback on node touch and we're explicitly disabling it... What's the motivation behind your PR?

**Comment by brightsider** [comment]
I don't need a touch animation when don't have children. I can add options to control this

**Comment by zaguiini** [comment]
So isn't it better to add a prop like `disableTouchOnLeaves` and then having this condition checked on the `disabled` prop at `TouchableOpacity` element?

**Comment by brightsider** [comment]
`disableTouchOnLeaves` should be a function or a boolean?

**Comment by zaguiini** [comment]
A function. A better name for it would be `shouldDisableTouchOnLeaf(node)` I think. WDYT?

This function should be optional and return `false` by default.

**Comment by brightsider** [comment]
It's looking better? :)

File: index.js [hunk 1]
```diff
@@ -106,6 +108,10 @@ class TreeView extends React.Component {
           }}
         >
           <TouchableOpacity
+            disabled={this.props.shouldDisableTouchOnLeaf({
+              [this.props.idKey]: node[this.props.idKey],
```
Comment (zaguiini) [inline index.js]: You should pass only `node` here

**Review by zaguiini — CHANGES_REQUESTED** [review]

**Comment by zaguiini** [comment]
Thanks!

**Comment by zaguiini** [comment]
Shipped on `v2.0.3`.

**Pull request merged**